### DISCUSSION TITLE
Self-nested modules not showing in Browser

### DISCUSSION THREAD

**Discussion by Tom-revere** [area17/twill#2463] [Q&A]
## Description

I have created a 'Pages' module that allows self nesting using [this](https://twillcms.com/docs/modules/nested-modules.html#content-self-nested-modules), however when creating a Browser block it only shows top level Pages.

### Steps to reproduce

1. Create self-nested modules using [this](https://twillcms.com/docs/modules/nested-modules.html#content-self-nested-modules) guide.
2. Create a parent page, and a child page.
3. Create a block with a browser field: 
`<x-twill::browser
    module-name="pages"
    name="page"
    label="Page"
    :max="1"
/>`
4. Add this block to a page

### Expected result

The browser should show all pages, regardless of whether they are a parent or a child page.

### Actual result

It only shows the parent pages in the Browser field

### Versions

Twill: 8.0.1

Laravel: 9.52.4

PHP: 8.0

**Comment by ifox** [comment]
Hi @Tom-revere, you can set `showOnlyParentItemsInBrowsers` to false in your controller:
https://github.com/area17/twill/blob/540fc45871f64df7f80823c20f951acf1b796443/src/Http/Controllers/Admin/NestedModuleController.php#L16

**Comment by Tom-revere** [comment]
Thanks @ifox that worked perfectly!

Just a suggestion, is it possible to indent the child pages in the Browser, so it looks similar to the Module listing table? As when you have a lot of pages, it could be hard to distinguish which are child pages, and which are not.

![image](https://github.com/area17/twill/assets/70330106/51fe9697-6f5c-4be3-8a2f-bc3ff294ac99)
Has no Indentation in Browser

![image](https://github.com/area17/twill/assets/70330106/a461a5f7-bc9f-46a7-8d69-ef249bd6a348)
Has indentation in the Module list

Thanks again!

**Comment by denis-gorin** [comment]
@Tom-revere, Hi
if it is still actual, you can make workaround
by redefine controller method `getBrowserTableData` of your nested module controller (ex. PageController)
this will output in browser pop-over `Parent Page / Child Page` - better then nothing

```
protected function getBrowserTableData($items)
    {
        $withImage = $this->moduleHas('medias');

        $indexData = $this->indexData($this->request);

        $isNested = false;
        if (isset($indexData['nested'])){
            $isNested = $indexData['nested'];
        }

        return $items->map(function ($item) use ($withImage, $items, $isNested) {
            $columnsData = Collection::make($this->browserColumns)->mapWithKeys(function ($column) use ($item) {
                return $this->getItemColumnData($item, $column);
            })->toArray();

            $name = $columnsData[$this->titleColumnKey];
            unset($columnsData[$this->titleColumnKey]);

            $parentName = '';
            if ($isNested && $item->parent_id) {
                $parent = $items->where('id', $item->parent_id)->first();
                $parentName = $parent ? $parent->title . ' / ' : '';
            }

            return [
                'id' => $this->getItemIdentifier($item),
                'parent_id' => $item->parent_id,
                'name' => $parentName . $name,
                'edit' => moduleRoute($this->moduleName, $this->routePrefix, 'edit', $this->getItemIdentifier($item)),
                'endpointType' => $this->repository->getMorphClass(),
            ] + $columnsData + ($withImage && ! array_key_exists('thumbnail', $columnsData) ? [
                'thumbnail' => $item->defaultCmsImage(['w' => 100, 'h' => 100]),
            ] : []);
        })->toArray();
    }
